### PR TITLE
Move Jenkins workflow triggering to event handlers

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

Move Jenkins workflow triggering to event handlers

### Description of the Change

This PR will remove the jenkins webhook and use handlers to trigger the corresponding jobs

### Benefits

Become parity with gitlab and bitbucket workflows

### CODE REVIEW

File: core/jazz_environment-event-handler/index.js [hunk 1]
```diff
@@ -31,454 +31,551 @@ var processedEvents = [];
...
-				.then(result => { return resolve(result); })
-				.catch(err => {
-					logger.error("processEventDeleteBranch Failed" + err);
-					return reject(err);
-				})
-		}
-
-	});
+  return new Promise((resolve, reject) => {
+    logger.error("event payload : " + JSON.stringify(eventPayload));
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_environment-event-handler/index.js]: done

File: core/jazz_environment-event-handler/index.js [hunk 2]
```diff
@@ -31,454 +31,551 @@ var processedEvents = [];
...
+    },
+    rejectUnauthorized: false
+  };
+
+  svcPayload.uri = apiEndpoint;
+  svcPayload.method = method;
+  if (payload) {
+    svcPayload.json = payload;
+  }
+  logger.error("payload :" + JSON.stringify(svcPayload));
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_environment-event-handler/index.js]: done

File: core/jazz_environment-event-handler/index.js [hunk 3]
```diff
@@ -31,454 +31,551 @@ var processedEvents = [];
...
+  return new Promise((resolve, reject) => {
+    logger.error("Trigger jenkins job:" + JSON.stringify(result));
+    var output = JSON.parse(result);
+    if (!output.data && !output.data.services && output.data.services.length > 0) {
+      logger.error("Service details not found in service catalog: " + JSON.stringify(output));
+      var error = handleError(failureCodes.PR_ERROR_5.code, failureCodes.PR_ERROR_5.message);
+      return reject(error);
+    }
+    var serviceDetails = output.data.services[0];
+    logger.error("service details : " + JSON.stringify(serviceDetails));
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_environment-event-handler/index.js]: done

File: core/jazz_environment-event-handler/index.js [hunk 4]
```diff
@@ -31,454 +31,551 @@ var processedEvents = [];
...
+
+    var buildQuery = `/buildWithParameters?token=${configData.JOB_TOKEN}&service_name=${serviceDetails.service}&domain=${serviceDetails.domain}&scm_branch=${payload.physical_id}`;
+    var authToken = "Basic " + new Buffer(configData.JENKINS_USER + ":" + configData.API_TOKEN).toString("base64");
+    var apiEndpoint = `${configData.JOB_BUILD_URL}${configData.BUILDPACKMAP[serviceDetails.type]}${buildQuery}`;
+    var svcPayload = getSvcPayload("POST", null, apiEndpoint, authToken);
+
+    processRequest(svcPayload)
+      .then(result => { logger.error("successfully deployment started."); return resolve(result); })
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_environment-event-handler/index.js]: done

File: core/jazz_environment-event-handler/index.js [hunk 5]
```diff
@@ -31,454 +31,551 @@ var processedEvents = [];
...
+  processEventUpdateEnvironment: processEventUpdateEnvironment,
+  processEventCreateBranch: processEventCreateBranch,
+  processEventInitialCommit: processEventInitialCommit,
+  getEnvironmentLogicalId: getEnvironmentLogicalId,
+  processBuild:processBuild,
+  triggerBuildJob: triggerBuildJob,
+  getServiceDetails: getServiceDetails,
+  processRequest: processRequest,
+  getSvcPayload: getSvcPayload
 }
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_environment-event-handler/index.js]: Sure

File: core/jazz_environment-event-handler/config/test-config.json [hunk 6]
```diff
@@ -1,26 +1,56 @@
...
-    "ENVIRONMENT_DELETE_STARTED_STATUS": "deletion_started",
-    "ENVIRONMENT_DELETE_FAILED_STATUS": "deletion_failed",
-    "ENVIRONMENT_DELETE_COMPLETED_STATUS": "archived",
-    "RANDOM_CHARACTERS": "0123456789abcdefghijklmnopqrstuvwxyz",
-    "RANDOM_ID_CHARACTER_COUNT": 10
+  "BASE_API_URL": "https://{conf-apikey}.execute-api.{conf-region}.amazonaws.com",
+  "ENVIRONMENT_API_RESOURCE": "/dev/jazz/environments",
+  "DELETE_ENVIRONMENT_API_RESOURCE": "/dev/jazz/delete-serverless-service",
+  "TOKEN_URL": "/dev/jazz/login",
+  "SERVICE_API_RESOURCE": "/prod/jazz/services",
```
Comment (suryajak) [inline core/jazz_environment-event-handler/config/test-config.json]: this should be non-prod one, perhaps: `/dev/jazz/services`

Comment (SiniWilson) [reply core/jazz_environment-event-handler/config/test-config.json]: done

**Pull request merged**